client.go

```diff
@@ -161,7 +178,31 @@ func (c *StatsdClient) send(stat string, format string, value interface{}) error
 	}
 	stat = strings.Replace(stat, "%HOST%", Hostname, 1)
 	format = fmt.Sprintf("%s%s:%s", c.prefix, stat, format)
-	_, err := fmt.Fprintf(c.conn, format, value)
+	metricValue := fmt.Sprintf(format, value)
+
+	var err error
+
+	if c.flushInterval != 0 {
+		if !c.timerRunning {
+			c.startTimer()
+		}
+
+		c.statsMutex.Lock()
```

Comment:
can we do `defer statsMutex.Unlock()` just in case of panic? (instead of Unlock() down below)